### PR TITLE
Update secret scope validation to include `/` characters

### PR DESCRIPTION
**Pull request by tylangesmith**
This PR fixes issue #888

### CODE REVIEW

File: access/resource_secret_acl_test.go [hunk 1]
```diff
@@ -79,6 +79,74 @@ func TestSecretsScopesAclsIntegration(t *testing.T) {
 	assert.NoError(t, err, err)
 }
 
+func TestSecretsScopesAclsIntegration_ScopeWithSlash(t *testing.T) {
```
Comment (nfx) [inline access/resource_secret_acl_test.go:82]: Test prefix must be "TestAcc"

Comment (nfx) [reply access/resource_secret_acl_test.go:82]: This test doesn't add value to provider - SDK layer doesn't perform validation, Terraform resource does.

Create a test in acceptance package instead and add a scope with a slash in name.

File: access/resource_secret_acl_test.go [hunk 2]
```diff
@@ -121,6 +211,26 @@ func TestResourceSecretACLRead_NotFound(t *testing.T) {
 	}.ApplyNoError(t)
 }
 
+func TestResourceSecretACLRead_ScopeWithSlash_NotFound(t *testing.T) {
```
Comment (nfx) [inline access/resource_secret_acl_test.go:214]: This test is not necessary

File: access/resource_secret_acl_test.go [hunk 3]
```diff
@@ -142,6 +252,27 @@ func TestResourceSecretACLRead_Error(t *testing.T) {
 	assert.Equal(t, "global|||something", d.Id(), "Id should not be empty for error reads")
 }
 
+func TestResourceSecretACLRead_ScopeWithSlash_Error(t *testing.T) {
```
Comment (nfx) [inline access/resource_secret_acl_test.go:255]: This test is not necessary as well - url encoding of a slash was tested by the first test added

File: access/resource_secret_acl_test.go [hunk 4]
```diff
@@ -174,6 +305,38 @@ func TestResourceSecretACLCreate(t *testing.T) {
 	assert.Equal(t, "global|||something", d.Id())
 }
 
+func TestResourceSecretACLCreate_ScopeWithSlash(t *testing.T) {
```
Comment (nfx) [inline access/resource_secret_acl_test.go:308]: Probably just keep this test, as it covers both read and create

**Review by nfx — CHANGES_REQUESTED** [review]
Please remove redundant unit tests and keep only one for creating. Url encoding and validation is the test focus, not already tested functionality.

**Review by nfx — APPROVED** [review]
Lgtm

**Comment by nfx** [comment]
@tylangesmith please rebase onto latest master

**Comment by tylangesmith** [comment]
Forgot to tag you this morning @nfx but I believe that should be rebased now

**Review by nfx — APPROVED** [review]

**Pull request merged**